### DISCUSSION TITLE
Not able to fetch the value using CTk.Entry

### DISCUSSION THREAD

**Discussion by vixnexh** [TomSchimansky/CustomTkinter#1320] [General]
Hey there! I need a very urgent help with this so that i can finish my tool. So im trying to use the Entry widget to get the value thats been entered and use it as a string to change it and use it for another funtion. But im trying to .get() from the Entry in multiple ways but none of them seem to work. Ive attached a sample code below

import customtkinter

app = customtkinter.CTk() 
app.geometry("400x400")
app.title("test_entry_placeholder.py")

str_var = customtkinter.StringVar()

def button_action():
    print(entry_1.get)
    
entry_1 = customtkinter.CTkEntry(app, placeholder_text="placeholder", textvariable=str_var)
entry_1.pack(pady=20)
print(entry_1.get())
print(str_var.get())
entry_2 = customtkinter.CTkEntry(app, placeholder_text="placeholder", textvariable=str_var)
entry_2.pack(pady=20)
entry_3 = customtkinter.CTkButton(app, text = "OK", command = button_action)
entry_3.pack(pady=30)
text = entry_1.get
print(text)
print(str_var.get())

app.mainloop()

**Comment by Akascape** [comment]
@vixnexh 
Use tkinter string var:
```python
import tkinter
str_var = tkinter.StringVar()
```
Then properly use the get method:
```python
text = entry_1.get()
print(text)
print(str_var.get())
```

(And please try to give error logs)

**Reply by vixnexh** [reply]
@Akascape 

Im getting this in the log
<bound method CTkEntry.get of <customtkinter.windows.widgets.ctk_entry.CTkEntry object .!ctkentry>>

For debug purpose i tried to use different methods to use the .get(). But when i type in any values in the entry box, i dont see anything being printed during run time

**Reply by Akascape** [reply]
@vixnexh You have to use `entry_1.get()`, not `entry_1.get`.

**Reply by vixnexh** [reply]
Tried it after making that correction. Still not able to see value being printed

**Reply by Akascape** [reply]
@vixnexh Don't print the value directly in the main loop, put the print function under any button function.
Here is a basic example:
```python
import customtkinter

app = customtkinter.CTk()

def button_action():
    print(str_var.get())
    print(entry_1.get())
    
str_var = customtkinter.StringVar()

entry_1 = customtkinter.CTkEntry(app, textvariable=str_var)
entry_1.pack(padx=10)

button = customtkinter.CTkButton(app, text="Print", command=button_action)
button.pack(padx=10, pady=10)

app.mainloop()
```
**The value will be printed only when you write something in the entry and click the button.**
